engine/step/step.go

```diff
@@ -264,6 +300,59 @@ func Run(st *Step, baseConfig map[string]json.RawMessage, values *values.Values,
 		case <-stopRunningSteps:
 			st.State = StateToRetry
 		default:
+			for _, hookName := range st.PreHooks {
+				h, err := getHook(hookName)
+				if err != nil {
+					st.Error = fmt.Sprintf("get hook %s failed: %v", h.Name, err)
+					st.State = StateFatalError
+					go noopStep(st, stepChan)
+					return
+				}
+
+				o, m, err := runner.Exec(h.Name, baseCfgRaw, h.Action.Configuration, ctx)
+				values.SetHookOutput(st.Name, h.Name, o)
+				values.SetHookMetadata(st.Name, h.Name, m)
+				if err != nil {
+					if errors.IsBadRequest(err) {
+						st.State = StateClientError
+					} else {
+						st.State = StateServerError
+					}
+					st.Error = err.Error()
+					go noopStep(st, stepChan)
+					return
+				} else if st.ResultValidate != nil {
+					if err := st.ResultValidate(st.Output); err != nil {
```

Comment:
ResultValidate is not declared on Hook structure. This code should happens on a ResultValidate structure existing inside the hook, and not on the step one.
Also, the body to validate if the one from the hook output, and not the step output.